syntax = "proto2";
package serviceCommon;

message KeyValue {
    required string key = 1;
    oneof value {
        int32 integerVal = 2;
        uint32 uIntegerVal = 3;
        float floatVal = 4;
        string stringVal = 5;
    }
}

message KeyValueList {
    required string category = 1;
    repeated KeyValue element = 2;
}

message upmuStructKeyValueList {
    required uint32 timeStamp = 1;   
    required string name = 2;
    repeated KeyValueList list = 3;
}
    
